src/generator/workers/src/typescript.rs

```diff
@@ -92,7 +90,56 @@ impl TypescriptValidatorGenerator {
     }
 }
 
-pub struct TypescriptWorkersGenerator;
+pub struct TypescriptWorkersGenerator {
+    domain: String,
+    root_path: String,
+    router: RouterTrie,
+}
+
+impl TypescriptWorkersGenerator {
+    pub fn new() -> Self {
+        Self {
+            domain: String::new(),
+            root_path: "api".to_string(),
+            router: RouterTrie::new("api"),
+        }
+    }
+
+    pub fn parse_domain(domain: String) -> (String, String) {
```

Comment:
parse domain is not dependent on a specific language, it should not be apart of the typescript generator